### PR TITLE
[347] handle 404 get previous transaction

### PR DESCRIPTION
**Pull request by toddlees**
api#347

### CODE REVIEW

File: front-end/src/app/shared/services/transaction.service.ts [hunk 1]
```diff
@@ -49,15 +50,25 @@ export class TransactionService implements TableListService<Transaction> {
...
-        .pipe(map((response) => getFromJSON(response)));
+        .get<HttpResponse<Transaction>>(
+          '/transactions/previous/',
+          {
+            transaction_id,
+            contact_id,
+            date: actionDateString,
+            aggregation_group,
+          },
+          [404]
```
Comment (dheitzer) [inline front-end/src/app/shared/services/transaction.service.ts]: May want to use for clarity: 

```
import { HttpStatusCode } from '@angular/common/http';
HttpStatusCode.NotFound

```

File: front-end/src/app/shared/interceptors/http-error.interceptor.ts [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
 export class HttpErrorInterceptor implements HttpInterceptor {
-  constructor(private store: Store) { }
+  constructor(private store: Store) {}
 
   intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
     return next.handle(request).pipe(
       catchError((error: HttpErrorResponse) => {
+        //pass on error code if allowed
+        if (request.context.get(ALLOW_ERROR_CODES).includes(error.status)) {
+          return of(new HttpResponse({ status: 404 }));
```
Comment (dheitzer) [inline front-end/src/app/shared/interceptors/http-error.interceptor.ts]: should this use error.status?

`new HttpResponse({ status: error.status })
`

Comment (toddlees) [reply front-end/src/app/shared/interceptors/http-error.interceptor.ts]: YES!

Comment (toddlees) [reply front-end/src/app/shared/interceptors/http-error.interceptor.ts]: nice catch

**Review by Elaine-Krauss-TCG — APPROVED** [review]

**Pull request merged**